partner_middlename/models/res_partner.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2023 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+from odoo.addons.partner_firstname import exceptions
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    middlename = fields.Char("Middle Name")
+
+    @api.model
+    def _get_computed_name(self, lastname, firstname, middlename=None):
+        """Compute the name combined with the second lastname too.
+
+        We have 2 lastnames, so lastnames and firstname will be separated by a
+        comma.
+        """
+        order = self._get_names_order()
+        names = list()
+        if order == "last_first":
+            if firstname:
+                names.append(firstname)
+            if middlename:
+                names.append(middlename)
+            if lastname:
+                names.append(lastname)
+        else:
+            if lastname:
+                names.append(lastname)
+            if names and firstname and order == "last_first_comma":
+                names[-1] = names[-1] + ","
+            if firstname:
+                names.append(firstname)
+            if middlename:
+                names.append(middlename)
+        return " ".join(names)
+
+    @api.depends("firstname", "lastname", "middlename")
+    def _compute_name(self):
+        """Write :attr:`~.name` according to splitted data."""
+        for partner in self:
+            partner.name = self._get_computed_name(
+                partner.lastname,
+                partner.firstname,
+                partner.middlename,
+            )
+
+    def _inverse_name(self):
```

Comment:
The inverse was not working on my tests - setting the Name should compute the First+Middle+Last.
Can you check please?